clang/lib/Headers/hlsl/hlsl_detail.h

```diff
@@ -45,9 +45,17 @@ template <typename T> struct is_arithmetic {
   static const bool Value = __is_arithmetic(T);
 };
 
+template <typename T> struct is_vector {
+  static const bool value = false;
+};
+
+template <typename T, int N> struct is_vector<vector<T, N>> {
+  static const bool value = true;
+};
+
 template <typename T, int N>
 using HLSL_FIXED_VECTOR =
-    vector<__detail::enable_if_t<(N > 1 && N <= 4), T>, N>;
+    vector<__detail::enable_if_t<(N >= 1 && N <= 4), T>, N>;
```

Comment:
Please revert this change. This  is going to have broader implications that you might think. we were trying to force vec1 down the scalar path.